spec/cart_spec.rb

```diff
@@ -0,0 +1,57 @@
+require "cart"
+
+describe Cart do
+
+  # TODO: Learn how to use doubles/specs/mocks instead of this:
+  let(:menu_item_1) { MenuItem.new(1, "Burger", "The best burger in the world", 100, "USD") }
+  let(:menu_item_2) { MenuItem.new(2, "Fries", "The best fries in the world", 50, "USD") }
+  let(:menu) { Menu.new([menu_item_1, menu_item_2]) }
+  let(:order_item_1) { OrderItem.new(menu, 0, 1) }
+  let(:order_item_2) { OrderItem.new(menu, 1, 2) }
+  let(:cart) { Cart.send(:new) } # See http://stackoverflow.com/a/23901644
+
+  it "is a kind of Cart" do
+    expect(cart).to be_a_kind_of(Cart)
+  end
+
+  context "a 'Burger' is added to the cart" do
+    before do
+      Cart.instance.add_item_to_cart(order_item_1) # Comment from Thomas: "Why use Cart.instance vs storing that instance in a variable?"
+    end
+    it "adds a 'Burger' to the cart" do
+      # See FOOTNOTE 1
+      expect(Cart.instance.cart[0].name).to eq("Burger")
+    end
+  end
+
+    context "an order of 'Fries' is added to the cart" do
+    before do
+      Cart.instance.add_item_to_cart(order_item_2)
+    end
+    it "adds an order of 'Fries' to the cart" do
+      expect(Cart.instance.cart[1].name).to eq("Fries")
+    end
+  end
```

Comment:
Lint/BlockAlignment: `end` at 34, 2 is not aligned with `context "an order of 'Fries' is added to the cart" do` at 27, 4